app/views/api/v2/layouts/permissions.json.rabl

```diff
@@ -0,0 +1,6 @@
+object @resource
+
+if params.has_key?(:include_permissions)
+  node(:can_edit) { |resource| authorizer.can?(:edit, resource) }
```

Comment:
Thank you so much @ares. That helped me so much.
However, I wasn't sure about the last part.
I think `authorizer` in `authorize_helper.rb` is taking care of setting the base collection set.
So I just passed to `authorized_for` the returned value from `authorizer`.

Let me know if I'm missing something.

Reply:
> I think authorizer in authorize_helper.rb is taking care of setting the base collection set.

Yes, that's true for hw models page and this use case. But this is a generic layout which people could use with whatever collections. Also note that hammer (foreman CLI) asks for 1 000 records at a time by default. Therefore all 1 000 records will be stored in [default authorizer collection](
https://github.com/theforeman/foreman/blob/db0b8c190abf17e14e6f5055c2c63c8cfdfbbe08/app/helpers/authorize_helper.rb#L30). Given our API supports pagination parameters, I wonder what's better, load all records to memory and verify permission once for potentially thousands of records, or verify permissions for each of such objects separately.

Anyway using authorizer like you do now already helped since instead of two separate caches, we have one now. There's still one problem for which I'll open a separate thread, in general you're not passing object to authorize.